### PR TITLE
Add e2e integration tests for postgres functions

### PR DESCRIPTION
**Pull request by PrasadG193**
## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by PrasadG193** [comment]
Steps to run E2E tests for RDS functions

File: pkg/blueprint/blueprints/rds-postgres-dump-blueprint.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    - func: ExportRDSSnapshotToLocation
+      name: exportSnapshot
+      args:
+        namespace: "rds-postgres-dump-test"
+        username: '{{ index .Secrets.dbsecret.Data "username" | toString }}'
+        password: '{{ index .Secrets.dbsecret.Data "password" | toString }}'
+        backupArtifactPrefix: test-postgresql-instance/postgres
+        instanceID:  "{{ .ArtifactsIn.backupInfo.KeyValue.instanceID }}"
+        snapshotID:  "{{ .ArtifactsIn.backupInfo.KeyValue.snapshotID }}"
+        dbEngine: "PostgreSQL"
```
Comment (PrasadG193) [inline pkg/blueprint/blueprints/rds-postgres-dump-blueprint.yaml]: Remove this

File: pkg/blueprint/blueprints/rds-postgres-dump-blueprint.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    phases:
+    - func: RestoreRDSSnapshot 
+      name: restoreSnapshots
+      args:
+        namespace: "rds-postgres-dump-test"
+        backupArtifactPrefix: test-postgresql-instance/postgres
+        instanceID:  "{{ .ArtifactsIn.backupInfo.KeyValue.instanceID }}"
+        backupID:  "{{ .ArtifactsIn.backupInfo.KeyValue.backupID }}"
+        username: '{{ index .Secrets.dbsecret.Data "username" | toString }}'
+        password: '{{ index .Secrets.dbsecret.Data "password" | toString }}'
```
Comment (PrasadG193) [inline pkg/blueprint/blueprints/rds-postgres-dump-blueprint.yaml]: Change the mechanism to pass secret ref

File: pkg/blueprint/blueprints/rds-postgres-dump-blueprint.yaml (hunk 1 above)
Comment (PrasadG193) [reply pkg/blueprint/blueprints/rds-postgres-dump-blueprint.yaml]: @tdmanv Done!

File: pkg/aws/rds/rds.go [hunk 3]
```diff
@@ -16,33 +16,39 @@ package rds
...
+	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/rds"
 	"github.com/pkg/errors"
 )
 
+const (
+	maxRetries      = 10
```
Comment (tdmanv) [inline pkg/aws/rds/rds.go]: This seems like a lot, is it requried

File: pkg/function/create_rds_snapshot.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	rdsCli, err := rds.NewClient(ctx, awsConfig, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create Snapshot
+	log.Print("Creating RDS snapshot", field.M{"SnapshotID": snapshotID})
+	_, err = rdsCli.CreateDBSnapshot(ctx, instanceID, snapshotID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to create snapshot")
```
Comment (tdmanv) [inline pkg/function/create_rds_snapshot.go]:
```suggestion
		return nil, errors.Wrap(err, "Failed to create snapshot")
```

**Comment by PrasadG193** [comment]
Configmap format:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: dbconfig
  namespace: rds-postgres-dump-test
data:
  postgres.database: postgres
  postgres.host: <rds-endpoint>
  postgres.instanceid: <rds-instance-id>
  postgres.secret: dbsecret
  postgres.user: <master-username>
```

**Pull request merged**